src/renderers/testing/ReactTestRendererFiber.js

```diff
@@ -0,0 +1,332 @@
+};
+
+var TestRenderer = ReactFiberReconciler({
+  getRootHostContext() {
+    return emptyObject;
+  },
+
+  getChildHostContext() {
+    return emptyObject;
+  },
+
+  prepareForCommit() : void {
+    // noop
+  },
+
+  resetAfterCommit() : void {
+    // noop
+  },
+
+  createInstance(
+    type : string,
+    props : Props,
+    rootContainerInstance : Container,
+    hostContext : Object,
+    internalInstanceHandle : Object,
+  ) : Instance {
+    return new TestComponent(type, props, rootContainerInstance);
+  },
+
+  appendInitialChild(parentInstance : Instance, child : Instance | TextInstance) : void {
+    parentInstance.appendChild(child);
+  },
+
+  finalizeInitialChildren(
+    testElement : Instance,
+    type : string,
+    props : Props,
+    rootContainerInstance : Container,
+  ) : boolean {
+    // console.log('finalizeInitialChildren');
+    // setInitialProperties(testElement, type, props, rootContainerInstance);
+    return false;
+  },
+
+  prepareUpdate(
+    testElement : Instance,
+    type : string,
+    oldProps : Props,
+    newProps : Props,
+    hostContext : Object,
+  ) : boolean {
+    return true;
+  },
+
+  commitUpdate(
+    instance : Instance,
+    type : string,
+    oldProps : Props,
+    newProps : Props,
+    rootContainerInstance : Container,
+    internalInstanceHandle : Object,
+  ) : void {
+    instance.update(type, newProps);
+  },
+
+  commitMount(
+    instance : Instance,
+    type : string,
+    newProps : Props,
+    rootContainerInstance : Object,
+    internalInstanceHandle : Object
+  ) : void {
+    // Noop
+  },
+
+  shouldSetTextContent(props : Props) : boolean {
+    return (
+      typeof props.children === 'string' ||
+      typeof props.children === 'number'
+    );
+  },
+
+  resetTextContent(testElement : Instance) : void {},
+
+  createTextInstance(
+    text : string,
+    rootContainerInstance : Container,
+    hostContext : Object,
+    internalInstanceHandle : Object
+  ) : TextInstance {
+    var inst = {
+      text : text,
+      id: instanceCounter++,
+      rootContainerInstance,
+      toJSON: () => isNaN(+inst.text) ? inst.text : +inst.text,
+    };
+    // Hide from unit tests
+    Object.defineProperty(inst, 'id', { value: inst.id, enumerable: false });
+    return inst;
+  },
+
+  commitTextUpdate(textInstance : TextInstance, oldText : string, newText : string) : void {
+    textInstance.text = newText;
+  },
+
+  appendChild(parentInstance : Instance | Container, child : Instance | TextInstance) : void {
+    parentInstance.appendChild(child);
+  },
+
+  insertBefore(
+    parentInstance : Instance | Container,
+    child : Instance | TextInstance,
+    beforeChild : Instance | TextInstance
+  ) : void {
+    parentInstance.insertBefore(beforeChild, child);
+  },
+
+  removeChild(parentInstance : Instance | Container, child : Instance | TextInstance) : void {
+    parentInstance.removeChild(child);
+  },
+
+  scheduleAnimationCallback: window.requestAnimationFrame,
```

Comment:
I think this will throw in non-jsdom environment. Can you make it `setTimeout`? It seems like it wouldn't be used anyway, but at least we shouldn't throw.